src/components/themed/EdgeTextField.js

```diff
@@ -1,20 +1,26 @@
 // @flow
 
 import * as React from 'react'
+import { TouchableOpacity, View } from 'react-native'
 // $FlowFixMe = OutlinedTextField is not recognize by flow
 import { type TextFieldProps, OutlinedTextField, TextField } from 'react-native-material-textfield'
+import AntDesignIcon from 'react-native-vector-icons/AntDesign'
 
 import { unpackEdges } from '../../util/edges.js'
-import { type ThemeProps, withTheme } from '../services/ThemeContext.js'
+import { type Theme, type ThemeProps, cacheStyles, withTheme } from '../services/ThemeContext.js'
 
 type EdgeTextFieldProps = {
-  fieldRef?: ?React.ElementRef<typeof OutlinedTextField>,
   marginRem?: number | number[]
 }
 
-type Props = EdgeTextFieldProps & ThemeProps & TextFieldProps
+type EdgeOutlinedTextFieldProps = {
+  fieldRef?: ?React.ElementRef<typeof OutlinedTextField>,
+  marginRem?: number | number[],
+  isClearable: boolean,
+  onClear: () => void
```

Comment:
This callback should not be needed. The clear button should simply clear the text value which would automatically call the `onChangeText` callback.

Reply:
This one is a bit tricky because of how the Transaction List search handles the filter. Transaction List only filters when the keyboard search button is pushed or the Input is blurred. It is not the same as the Wallet List where every type you do it filters the list. Why the Transaction List filters like that is because getTransaction filter is expensive as you discussed before. And suggest this might be the most optimal way.

Now, here are the scenarios when pressing the clear button if it only changes the input state and no callback:
- Keyboard is up and you type stuff, you press clear anytime. It just erase the text, the list is not affected until you dismiss the keyboard. No problem
- Keyboard is down, the list is filtered already, and you press clear. This will clear the text BUT IT WON'T RE-FILTER the list. Meaning to clear the filter again, you need to open up keyboard/focus the input and press search button to drop the keyboard, thus trigger the filter after.

No callbacks are viable if the Transaction List is filtered per type on the keyboard. Or if the keyboard down scenario is ok